mrobosub_planning/src/umrsm.py

```diff
@@ -56,6 +70,23 @@ def handle(self) -> Optional[NamedTuple]:
     def is_valid_income_type(cls, outcome_type: Type[NamedTuple]) -> bool:
         return True
 
+    @classmethod
+    def with_params(cls, **kwargs: Any) -> Type['State']:
+        num_unexpected = 0
+        for k in kwargs:
+            if not hasattr(cls, k):
+                warnings.warn(f"overriding parameter {k}, which is not defined on {cls}", stacklevel=2)
+                num_unexpected += 1
+        overrides: dict = kwargs.get('_param_overrides', {}).copy()
+        overrides.update(kwargs)
```

Comment:
Could this potentially lead to nesting within the _param_overrides of just more _param_overrides

Reply:
Idts, but you made me realize what I had was wrong

Reply:
is the _param_overrides just for the printing of the rendered representation if so this all seems good to me

Reply:
Yeah it's just for the better prints